server/src/main/java/org/opensearch/search/approximate/ApproximatePointRangeQuery.java

```diff
@@ -114,21 +117,38 @@ public void visit(QueryVisitor visitor) {
 
     @Override
     public final ConstantScoreWeight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float boost) throws IOException {
+        // Pre-compute the effective bounds considering search_after
+        final byte[] effectiveLower = computeEffectiveLowerBound();
+        final byte[] effectiveUpper = computeEffectiveUpperBound();
+
+        this.pointRangeQuery = new PointRangeQuery(
```

Comment:
I have to re-instantiate this `PointRangeQuery` as there is a logic in `scorerSupplier` to use `pointRangeQueryWeight.scorerSupplier` when the size requested is greater than the segment docs. 
```
if (size > values.size()) {
    return pointRangeQueryWeight.scorerSupplier(context);
 }
 ```
 Without the re-instantiate, it will continue to use the original range bounds and not the new computed search_after range bounds which leads to incorrect search results.
 
 I'm open for thoughts here.